### DISCUSSION TITLE
[ALL HANDS] switch to OpenStreetMap

### DISCUSSION THREAD

**Discussion by anandijain** [free2pee/free2pee#8] [Ideas]
I want to hear from each and every one of you @acoh64 @michaelfortunato @free2pee/kernel-team 

https://mapcomplete.osm.be/toilets.html?z=1&lat=0.35156&lon=0&language=en#welcome
https://github.com/openstreetmap/osm2pgsql
https://github.com/pietervdvn/MapComplete

It seems like we might be able to bootstrap with this and not be BEHOLDEN to GOOGLE. 
![image](https://user-images.githubusercontent.com/33790004/219426200-00cee563-59c2-4626-a5c1-3c1273117be0.png)

**Comment by anandijain** [comment]
this is actually what we need to do, we need to just be a very good frontend to OSM and then they can manage the database and we just help crowdsource the info to them. we will still need all the infra and itll be fun to hack around in rust. but this is way better than using google

**Reply by acoh64** [reply]
That interface and graphics are not as nice as google though.

**Reply by anandijain** [reply]
personally im okay with that, but we could render Google Maps but then add in the OSM annotations.

but look at this https://www.openstreetmap.org/node/4817053508 it gives us a semantic link to which company owns the place (https://en.wikipedia.org/wiki/en:Target%20Corporation?uselang=en) which means our ability to inference can take into account company specific policy.

plus no more bullshit scraping, we can just download the planet

**Reply by anandijain** [reply]
https://wiki.openstreetmap.org/wiki/Tag:amenity=toilets?uselang=en look at their implementation of bathroom data annotations

**Reply by anandijain** [reply]
![image](https://user-images.githubusercontent.com/33790004/219432097-5e904f8a-a1b6-476f-b9e0-6062150e92c5.png)
 the public health angle is huge

**Reply by acoh64** [reply]
we could also use that to do the scraping (so we can use the links and not have to do so many hack-y api calls to google) but then use it with google maps. I think if people are actually going to use this is has to have a super nice interface.

**Reply by anandijain** [reply]
i hear you but we need to use this. the scalability of making api calls to google is extremely limited. plus a huge waste of our time. 

our goal is to get you enough data to start exploring and testing models. the way to do this is by downloading the planet.osm data.

i agree that it needs to be slick. will you test out https://wiki.openstreetmap.org/wiki/Software/Mobile the mobile apps for OSM and share your thoughts

**Reply by acoh64** [reply]
yeah I agree that we should use this to get the data. Ill try that out and let you know. need to implement social media portion of it too.

**Reply by anandijain** [reply]
yeah they have a lot of that though, we can just do "Sign in with OpenStreetMaps" (@michaelfortunato take note), you can see that here on the left https://mapcomplete.osm.be/toilets.html?z=1&lat=0.35156&language=en#welcome 

here is the API docs https://wiki.openstreetmap.org/wiki/API 

we should be able to add on top of the user database of OSM. see an example of my user here https://www.openstreetmap.org/user/anandjain it has some but not all of the features we want.

**Comment by anandijain** [comment]
https://planet.openstreetmap.org/ @michaelfortunato full database downloads

**Reply by anandijain** [reply]
>Planet.osm is the OpenStreetMap data in one file: all the [nodes, ways and relations](https://wiki.openstreetmap.org/wiki/Elements) that make up our map. A new version is released every week. It's a big file (on 2023-02-01, the plain OSM XML variant takes over 1696.6 GB when uncompressed from the 123.4 GB bzip2-compressed or 67.3 GB [PBF](https://wiki.openstreetmap.org/wiki/PBF_Format)-compressed downloaded data file).

from https://wiki.openstreetmap.org/wiki/Planet.osm

**Comment by anandijain** [comment]
https://www.openstreetmap.org/api/0.6/node/4360543295 here is an example of the central square public bathroom XML

here is the link to it on mapcomplete: https://mapcomplete.osm.be/toilets.html?z=21&lat=42.36546&lon=-71.1041&language=en#node/4360543295

**Comment by anandijain** [comment]
while there is a proposal process for asking for new tags, https://wiki.openstreetmap.org/wiki/Proposal_process, I imagine we can just come up with our own tagging scheme (https://wiki.openstreetmap.org/wiki/Any_tags_you_like ) and then decide if we want to propose them as official tags

**Reply by anandijain** [reply]
>There are many projects using OSM tags in some way: editors, routers, maps, data extraction tools, ... Taginfo can get information from those projects about their use of OSM tags through 'project files'. The taginfo database keeps the information which projects use which keys and tags (optionally with descriptions and/or links to the documentation of the project). This information is then shown by taginfo on each of the key/tag pages in the 'Projects' tab


from  https://wiki.openstreetmap.org/wiki/Taginfo/Projects 

so there is a common description of TagInfo views (like open toilet map)
A list of current projects can be found https://taginfo.openstreetmap.org/projects or on MapComplete
